pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -276,19 +297,30 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 		}
 	}()
 
-	if p.options.FullPhysicalCPUsOnly && ((numCPUs % p.topology.CPUsPerCore()) != 0) {
-		// Since CPU Manager has been enabled requesting strict SMT alignment, it means a guaranteed pod can only be admitted
-		// if the CPU requested is a multiple of the number of virtual cpus per physical cores.
-		// In case CPU request is not a multiple of the number of virtual cpus per physical cores the Pod will be put
-		// in Failed state, with SMTAlignmentError as reason. Since the allocation happens in terms of physical cores
-		// and the scheduler is responsible for ensuring that the workload goes to a node that has enough CPUs,
-		// the pod would be placed on a node where there are enough physical cores available to be allocated.
-		// Just like the behaviour in case of static policy, takeByTopology will try to first allocate CPUs from the same socket
-		// and only in case the request cannot be sattisfied on a single socket, CPU allocation is done for a workload to occupy all
-		// CPUs on a physical core. Allocation of individual threads would never have to occur.
-		return SMTAlignmentError{
-			RequestedCPUs: numCPUs,
-			CpusPerCore:   p.topology.CPUsPerCore(),
+	if p.options.FullPhysicalCPUsOnly {
+		CPUsPerCore := p.topology.CPUsPerCore()
+		if (numCPUs % CPUsPerCore) != 0 {
+			// Since CPU Manager has been enabled requesting strict SMT alignment, it means a guaranteed pod can only be admitted
+			// if the CPU requested is a multiple of the number of virtual cpus per physical cores.
+			// In case CPU request is not a multiple of the number of virtual cpus per physical cores the Pod will be put
+			// in Failed state, with SMTAlignmentError as reason. Since the allocation happens in terms of physical cores
+			// and the scheduler is responsible for ensuring that the workload goes to a node that has enough CPUs,
+			// the pod would be placed on a node where there are enough physical cores available to be allocated.
+			// Just like the behaviour in case of static policy, takeByTopology will try to first allocate CPUs from the same socket
+			// and only in case the request cannot be sattisfied on a single socket, CPU allocation is done for a workload to occupy all
+			// CPUs on a physical core. Allocation of individual threads would never have to occur.
+			return SMTAlignmentError{
+				RequestedCPUs: numCPUs,
+				CpusPerCore:   CPUsPerCore,
+			}
+		}
+		availablePhysicalCPUs := p.getAvailablePhysicalCPUs(s).Size()
+		if numCPUs > availablePhysicalCPUs {
```

Comment:
Let's add a comment (pretty much what we have in the commit message) to explain the rationale behind this check here.

Reply:
sure thing. Done.